packages/api/src/external/carequality/document/create-outbound-document-query-req.ts

```diff
@@ -9,6 +9,105 @@ import { doesGatewayNeedDateRanges } from "@metriport/core/external/carequality/
 const SUBJECT_ROLE_CODE = "106331006";
 const SUBJECT_ROLE_DISPLAY = "Administrative AND/OR managerial worker";
 
+function buildRequest({
+  requestId,
+  cxId,
+  now,
+  user,
+  initiator,
+  externalGateway,
+  patient,
+  dateFrom,
+  dateTo,
+}: {
+  requestId: string;
+  cxId: string;
+  now: dayjs.Dayjs;
+  user: string;
+  initiator: HieInitiator;
+  externalGateway: CQLink;
+  patient: Patient;
+  dateFrom?: string;
+  dateTo?: string;
+}): OutboundDocumentQueryReq {
+  const request: OutboundDocumentQueryReq = {
+    id: requestId,
+    cxId: cxId,
+    timestamp: now.toISOString(),
+    samlAttributes: {
+      subjectId: user,
+      subjectRole: {
+        code: SUBJECT_ROLE_CODE,
+        display: SUBJECT_ROLE_DISPLAY,
+      },
+      organization: initiator.name,
+      organizationId: initiator.oid,
+      homeCommunityId: initiator.oid,
+      purposeOfUse: createPurposeOfUse(),
+    },
+    gateway: {
+      homeCommunityId: externalGateway.oid,
+      url: externalGateway.url,
+    },
+    externalGatewayPatient: {
+      id: externalGateway.patientId,
+      system: externalGateway.systemId,
+    },
+    patientId: patient.id,
+  };
+
+  if (dateFrom && dateTo) {
+    request.documentCreationDate = {
+      dateFrom: dateFrom,
+      dateTo: dateTo,
+    };
+  }
+
+  return request;
+}
+
+function buildRequestsWithDateRanges({
+  requestId,
+  cxId,
+  now,
+  user,
+  initiator,
+  externalGateway,
+  patient,
+}: {
+  requestId: string;
+  cxId: string;
+  now: dayjs.Dayjs;
+  user: string;
+  initiator: HieInitiator;
+  externalGateway: CQLink;
+  patient: Patient;
+}): OutboundDocumentQueryReq[] {
+  const requests = [];
+  const length = 10;
+  for (let i = 0; i < length; i++) {
+    const dateTo = now.subtract(i * 6, "month").toISOString();
+    const dateFrom =
+      i !== length - 1
+        ? now.subtract((i + 1) * 6, "month").toISOString()
+        : now.subtract(25, "year").toISOString();
```

Comment:
25 years seem a pretty important threshold to be "hidden" in the middle of this code. Can we move it to a constant on the top of this file, please?